src/main/java/core/basesyntax/strategy/handlers/PurchaseHandler.java

```diff
@@ -0,0 +1,15 @@
+package core.basesyntax.strategy.handlers;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.FruitTransaction;
+
+public class PurchaseHandler implements OperationHandler {
+    @Override
+    public void operate(FruitTransaction fruitTransaction) {
+        int newAmount = Storage.storage.get(fruitTransaction.getFruit()) - fruitTransaction.getQuantity();
+        if (newAmount < 0) {
+            throw new RuntimeException("Can't do purchase, because amount will be less than 0");
```

Comment:
Uninformative message. I wand to buy 10 bananas, but the shop has only 5, how would I know how many bananas can I buy?